deployer/billing.py

```diff
@@ -0,0 +1,278 @@
+
+
+# FIXME: This doesn't actually work yet correctly
+def validate_billing_export():
+    # Get billing accounts info
+    with open(HERE.joinpath("config/billing-accounts.yaml")) as f:
+        accounts = yaml.load(f)
+
+    # Create a client
+    billing_client = billing_v1.CloudBillingClient()
+
+    managed_billing_accounts = {
+        a["id"] for a in accounts["gcp"]["billing_accounts"] if a["managed"]
+    }
+    logging_client = ConfigServiceV2Client()
+
+    # Handle the response
+    for ba in managed_billing_accounts:
+
+        # Make the request
+        response = billing_client.get_billing_account(
+            name=f"billingAccounts/{ba}",
+        )
+        if not response.open_:
+            print(f"Billing account {ba} is closed!")
+            sys.exit(1)
+
+        print(f"trying for {ba}")
+
+        # Make the request
+        sinks = list(logging_client.list_sinks(parent=f"billingAccounts/{ba}"))
+
+        # Handle the response
+        print(sinks)
+
+
+def month_validate(month_str: str):
+    """
+    Validate passed string matches YYYY-MM format.
+
+    Returns values in YYYYMM format, which is used by bigquery
+    """
+    match = re.match(r"(\d\d\d\d)-(\d\d)", month_str)
+    if not match:
+        raise typer.BadParameter(f"{month_str} should be formatted as YYYY-MM")
+    return f"{match.group(1)}{match.group(2)}"
+
+
+class CostTableOutputFormats(Enum):
+    """
+    Output formats supported by the generate-cost-table command
+    """
+
+    terminal = "terminal"
+    google_sheet = "google-sheet"
+
+
+@app.command()
+def generate_cost_table(
+    start_month: str = typer.Option(
+        (datetime.utcnow().replace(day=1) - timedelta(days=1)).strftime("%Y-%m"),
+        help="Starting month (as YYYY-MM) to produce cost data for. Defaults to last invoicing month.",
+        callback=month_validate,
+    ),
+    end_month: str = typer.Option(
+        datetime.utcnow().replace(day=1).strftime("%Y-%m"),
+        help="Ending month (as YYYY-MM) to produce cost data for. Defaults to current invoicing month",
+        callback=month_validate,
+    ),
+    output: CostTableOutputFormats = typer.Option(
+        CostTableOutputFormats.terminal,
+        help="Where to output the cost table to",
+    ),
+    google_sheet_url: str = typer.Option(
+        "https://docs.google.com/spreadsheets/d/1URYCMap-Lxm4e_pAAC3Esxda7tZzRhCS6d85pxUiVQs/edit#gid=0",
+        help="Write to given Google Sheet URL. Used when --output is google-sheet. billing-spreadsheet-writer@two-eye-two-see.iam.gserviceaccount.com should have Editor rights on this spreadsheet.",
+    ),
+):
+
+    with open(HERE.joinpath("config/billing-accounts.yaml")) as f:
+        accounts = yaml.load(f)
+
+    client = bigquery.Client()
+
+    rows = []
+
+    for a in accounts["gcp"]["billing_accounts"]:
+        if "bigquery" not in a:
+            print(
+                f'No bigquery table specified for billing account {a["id"]}, skipping',
```

Comment:
the account IDs are a little opaque as a message, when running this and seeing this message it's unclear whether I should do something. If this is diagnostic perhaps using logging at a higher log level rather than printing.